t_sne_roots/t_sne_roots.ipynb

```diff
@@ -0,0 +1,156 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "286d2e98-9f33-4e1e-bba4-f526360dfc59",
+   "metadata": {},
+   "source": [
+    "# Interactive t-SNE ROOTS dimensionality reduction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3d774823",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import datashader as ds\n",
+    "import colorcet as cc\n",
+    "\n",
+    "import holoviews as hv\n",
+    "from holoviews.operation.datashader import datashade, rasterize, dynspread\n",
+    "import panel as pn\n",
+    "\n",
+    "pn.extension()\n",
+    "hv.extension('bokeh')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9deb58ad-1c26-440c-a622-80a79159102c",
+   "metadata": {},
+   "source": [
+    "\n",
+    "*Visualization of t-Distributed Stochastic Neighbor Embedding of ROOTS language Corpus.*\n",
+    "\n",
+    "Trained data from Christopher Akiki (dataset `cakiki/roots-tsne-data` on Hugging Face "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c46d5abb-2a8c-4533-9e30-9970bf20e015",
+   "metadata": {},
+   "source": [
+    "## Load the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6bf4b1c8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tsne_embedding = pd.read_parquet('./data/train-00000-of-00001-9a434d9cf7fd233e.parquet')\n",
+    "df = pd.DataFrame(data=tsne_embedding, columns=['x','y','language'])\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d56df973-74b8-44af-a80b-cda8d304b4c6",
+   "metadata": {},
+   "source": [
+    "## Datashade and add hover layer\n",
+    "\n",
+    "Note that until client-side colormixing is supported, the `datashade` operation is used with an invisible hover layer overlaid on top."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b3f9e6b7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "points = hv.Points(df, ['x','y'], ['language'])\n",
+    "op = datashade(points, color_key=cc.glasbey_light,\n",
```

Comment:
How many distinct languages are there, and are they sorted by popularity? At a glance it looks like 20 or so main categories, and if so Category20 might give more vibrant colors, or maybe glasbey_category10 if there are 20 main ones but then lots of rare categories.